frontend/src/components/pages/TranslationPage.tsx

```diff
@@ -89,77 +91,77 @@ const TranslationPage = () => {
     );
   };
 
+  function saveVersionState() {
+    if (
+      versionHistoryStack.Undo.length > 0 &&
+      JSON.stringify(translatedStoryLines) !==
+        JSON.stringify(
+          versionHistoryStack.Undo[versionHistoryStack.Undo.length - 1],
+        ) &&
+      versionHistoryStack.Undo.length < MAX_STACK_SIZE
+    ) {
+      setVersionHistoryStack({
+        Undo: [
+          ...deepCopy(versionHistoryStack.Undo),
+          deepCopy(translatedStoryLines),
+        ],
+        Redo: [],
+      });
+    }
+  }
+
+  useEffect(() => {
```

Comment:
interesting approach - did you consider extending `Autosave` code/functionality? How did that go over? @e-wai might have some more ideas as she implemented it

Reply:
I did consider that, but ended up not doing that because the debouncing on Autosave doesn't feel intuitive because of the long pause required for the autosave to kick in